install/installer/pkg/components/server/deployment.go

```diff
@@ -26,7 +26,37 @@ import (
 func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
 	labels := common.DefaultLabels(Component)
 
-	configHash, err := common.ObjectHash(configmap(ctx))
+	var hashObj []runtime.Object
+	if objs, err := configmap(ctx); err != nil {
+		return nil, err
+	} else {
+		hashObj = append(hashObj, objs...)
+	}
+
+	hashObj = append(hashObj, &corev1.Pod{
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Env: []corev1.EnvVar{
+						{
+							Name: "DATABASE_TYPE",
+							Value: func() string {
+								if pointer.BoolDeref(ctx.Config.Database.InCluster, false) {
```

Comment:
I understand that it's not required, but: would it make sense to share code between all DB-dependant deployments? :thinking:

Reply:
wrt duplication: Yeah, in this case it might actually be counter-productive, as the only dependency is within the deployment itself. If we changed a single function, we might inadvertently change other as well.

I think what I'm missing most is a method name/comment that conveys the intention: what we do, and why these values are the important ones to establish identity.

Reply:
Yeah, I'll happily add a comment

Reply:
Comment added to both instances